uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1033.php

```diff
@@ -0,0 +1,131 @@
+
+  /**
+   * Migrates message templates from managed entities to upgraders
+   *
+   * @return bool
+   */
+  public function upgrade_1033() {
+    $this->up1033_removeReferenceToTemplateEntities();
+    $this->up1033_fixTemplates();
+
+    return TRUE;
+  }
+
+  /**
+   * Removes references to managed email template entities
+   */
+  private function up1033_removeReferenceToTemplateEntities() {
+    $query = 'DELETE FROM civicrm_managed WHERE `entity_type` = "MessageTemplate"
+AND module = "uk.co.compucorp.civicrm.hrleaveandabsences"';
+
+    CRM_Core_DAO::executeQuery($query);
+  }
+
+  /**
+   * Initiates checks for each message template as system workflow template
+   */
+  private function up1033_fixTemplates() {
+    $templateName = 'msg_tpl_workflow_leave';
+    $optionValues = [
+      [
+        'option_group_id' => $templateName,
+        'name' => 'civihr_leave_request_notification',
+      ],
+      [
+        'option_group_id' => $templateName,
+        'name' => 'civihr_toil_request_notification',
+      ],
+      [
+        'option_group_id' => $templateName,
+        'name' => 'civihr_sickness_record_notification',
+      ]
+    ];
+
+    foreach ($optionValues as $optionValue) {
+      $result = civicrm_api3('OptionValue', 'get', [
+        'option_group_id' => $optionValue['option_group_id'],
+        'name' => $optionValue['name'],
+      ]);
+      $this->up1033_checkTemplate(array_shift($result['values']));
+    }
+  }
+
+  /**
+   * Checks if template exist as system workflow and initiate fixes if not
+   *
+   * @param Array $optionValue
+   */
+  private function up1033_checkTemplate($optionValue) {
+    $templateResult = civicrm_api3('MessageTemplate', 'get', [
+      'return' => ['id', 'msg_title', 'msg_subject', 'msg_text', 'msg_html'],
+      'msg_title' => $optionValue['label'],
+    ]);
+    if ($templateResult['count'] === 0) {
+      $this->up1033_setupSystemWorkflowTemplate($optionValue);
+    }
+
+    if ($templateResult['count'] === 1) {
+      $this->up1033_moveTemplatesToSystemWorkflow(array_shift($templateResult['values']), $optionValue);
+    }
+  }
+
+  /**
+   * Migrates existing message templates to system workflow
+   *
+   * @param Array $messageTemplate
+   * @param Array $optionValue
+   */
+  private function up1033_moveTemplatesToSystemWorkflow($messageTemplate, $optionValue) {
+    civicrm_api3('MessageTemplate', 'create', [
+      'workflow_id' => $optionValue['id'],
+      'is_default' => 1,
+      'is_reserved' => 0,
+      'id' => $messageTemplate['id'],
+    ]);
+
+    $this->up1033_duplicateTemplateAsReserved($messageTemplate, $optionValue['id']);
+  }
+
+  /**
+   * Sets up email message template as system workflow template
+   *
+   * @param Array $optionValue
+   */
+  private function up1033_setupSystemWorkflowTemplate($optionValue) {
+    $emailTemplates = CRM_Core_Resources::singleton()->getPath('uk.co.compucorp.civicrm.hrleaveandabsences')
```

Comment:
@davialexandre I need your advice on which directory the renamed template should be or better way of implementing this.

Reply:
Move it a `message_templates` folder in the root folder of the L&A extension.

Note that this is far from ideal. The best thing here would be to probably split that file into smaller ones and maybe even use smarty templates for the html there, but given the amount of time around spend on this ticket, let's not do this now.